typescript/introduction/tsc-data-types-ii/tsc-enum.md

```diff
@@ -0,0 +1,105 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Enum
+
+---
+## Content
+
+An `enum` is used to define a set of named constants.
+
+```ts
+// numerical enum
+enum SomeData {
+  A = 111,
+  B = 222,
+  C = 333
+}
+
+let foo: SomeData = SomeData.A;
+console.log(foo);
+// 111
+```
+
+If we omit the values, the enum will assign values from `0` onwards:
+
+```ts
+// numerical enum
+enum SomeData {
+  A, // auto-assigned to 0
+  B, // auto-assigned to 1
+  C  // auto-assigned to 2
+}
+
+let foo: SomeData = SomeData.A;
+console.log(foo);
+// 0
+```
+
+You can also set the starting value from which to start the value sequence:
+
+```ts
+enum OtherData {
+  A = 1,
+  B, // auto-assigned to 2
+  C = 50,
+  D, // auto-assigned to 51
+}
+
+let foo: OtherData = OtherData.B;
+console.log(foo);
+// 2
+```
+
```

Comment:
let's also show how we'd encounter an error usinig an enum type?

Reply:
What came to mind here? Do you maybe have a suggestion?

Reply:
have a function that accepts only a value of 'a' or 'b' then call it with 'c' to illustrate how enum guards